src/finn/custom_op/fpgadataflow/rtl/streamingfifo_rtl.py

```diff
@@ -95,12 +95,13 @@ def generate_hdl(self, model, fpgapart, clk):
         code_gen_dict["$TOP_MODULE_NAME$"] = topname
         # make instream width a multiple of 8 for axi interface
         in_width = self.get_instream_width_padded()
-        count_width = int(self.get_nodeattr("depth") - 1).bit_length()
+        count_width = int(self.get_nodeattr("depth") + 1).bit_length()
         code_gen_dict["$COUNT_RANGE$"] = "[{}:0]".format(count_width - 1)
         code_gen_dict["$IN_RANGE$"] = "[{}:0]".format(in_width - 1)
         code_gen_dict["$OUT_RANGE$"] = "[{}:0]".format(in_width - 1)
         code_gen_dict["$WIDTH$"] = str(in_width)
         code_gen_dict["$DEPTH$"] = str(self.get_nodeattr("depth"))
+        code_gen_dict["$COUNT_WIDTH$"] = count_width
```

Comment:
If the `count_width` configuration is dropped above, this substitution will become obsolete.

Reply:
`countwidth` is not template parameter anymore, deleted from code generation.